src/codeu/chat/util/Tokenizer.java

```diff
@@ -1,12 +1,134 @@
 
+import com.sun.xml.internal.ws.policy.privateutil.PolicyUtils;
+
+import java.io.IOException;
+
 public final class Tokenizer {
 
-    public Tokenizer(String source) {
+    //stores the final "token" form of the input
+    private StringBuilder token;
+
+    //the input being processed/tokenized
+    private String source;
 
+    //index where current character being read is located
+    private int at;
+
+    public Tokenizer(String source) {
+        this.source = source;
+        at = 0;
+        token = new StringBuilder();
     }
 
+    /**
+     * Tokenizes the source String by turning each group of characters
+     * separated by white space into a String. Returns the next group of
+     * characters in the source String for every method call.
+     *
+     * @return  String      Next group of characters without white space in the source String.
+     * @throws IOException  May throw an IO Exception through calling the read() method.
+     */
     public String next() throws IOException {
+        //ignores all whitespace at beginning of the source String
+        while (remaining() > 0 && Character.isWhitespace(peek())){
+            read();
+        }
+        //returns null if there are no more characters in the String
+        if (remaining() <= 0 ){
+            return null;
+        }
+        // if the source String is surrounded by quotes
+        else if (peek() == '"'){
+           return readWithQuotes();
+        }
+        else {
+           return readWithNoQuotes();
+        }
+    }
+
+    /**
+     * Returns the next group of characters in the source String
+     * surrounded by quotes for every method call.
+     *
+     * @return  String      Next group of characters without white space in the source String.
+     * @throws IOException  May throw an IO Exception through calling the read() method or if
+     *                      String does not have a leading quotation mark.
+     */
+    private String readWithQuotes() throws IOException {
+        //clearing StringBuilder that will hold returned String
+        token.setLength(0);
+        //ensures first character is quotation, also allows next non-quote character to be read next
+        if (read() != '"'){
+            throw new IOException("String is not surrounded by quotes!");
+        }
+        //appends characters until a closing quotations mark is reached
+        while (remaining() > 0 && peek() != '"'){
+            token.append(read());
+        }
+        //reads the closing quotation mark
+        read();
+        return token.toString();
+    }
+
+    /**
+     * Returns the next group of characters in the source String
+     * with no surrounding quotes for every method call.
+     *
+     * @return  String      Next group of characters without white space in the source String.
+     * @throws IOException  May throw an IO Exception through calling the read() method.
+     */
+    private String readWithNoQuotes() throws IOException {
```

Comment:
Great! This will also help me become more familiar with the use of Lambda Expressions.